cmd/neofs-cli/modules/object/util.go

```diff
@@ -483,3 +480,72 @@ func collectObjectRelatives(ctx context.Context, cmd *cobra.Command, cli *client
 
 	return chain
 }
+
+type payloadWriter struct {
+	payload []byte
+}
+
+func (pw *payloadWriter) Write(p []byte) (n int, err error) {
+	pw.payload = append(pw.payload, p...)
+	return len(p), nil
+}
+
+func getChildrenFromV2Link(ctx context.Context, cmd *cobra.Command, cli *client.Client, key *ecdsa.PrivateKey, linkID oid.ID) ([]oid.ID, bool) {
+	var addrObj oid.Address
+	var prmGet internal.GetObjectPrm
+	pw := &payloadWriter{}
+
+	addrObj.SetObject(linkID)
+	prmGet.SetClient(cli)
+	prmGet.SetPrivateKey(*key)
+	prmGet.SetAddress(addrObj)
+	prmGet.SetRawFlag(false)
+	prmGet.SetPayloadWriter(pw)
+	Prepare(cmd, &prmGet)
+
+	_, err := internal.GetObject(ctx, prmGet)
+	if err == nil {
+		var link object.Link
+		err = link.Unmarshal(pw.payload)
+		if err != nil {
+			common.PrintVerbose(cmd, "failed to get linking object's header: %w", err)
+			return nil, false
+		}
+
+		res := make([]oid.ID, 0, len(link.Objects()))
+		for _, child := range link.Objects() {
+			res = append(res, child.ObjectID())
+		}
+
+		common.PrintVerbose(cmd, "Received split members from the linking object: %v", res)
+
+		// include linking object
+		return append(res, linkID), true
+	}
+
+	return nil, false
+}
+
+func getChildrenFromV1Link(ctx context.Context, cmd *cobra.Command, cli *client.Client, key *ecdsa.PrivateKey, linkID oid.ID) ([]oid.ID, bool) {
+	var addrObj oid.Address
+	var prmHead internal.HeadObjectPrm
+
+	prmHead.SetClient(cli)
+	prmHead.SetPrivateKey(*key)
+	addrObj.SetObject(linkID)
+	prmHead.SetAddress(addrObj)
+	prmHead.SetRawFlag(false)
+	Prepare(cmd, &prmHead)
+
+	res, err := internal.HeadObject(ctx, prmHead)
+	if err == nil {
+		children := res.Header().Children()
+
+		common.PrintVerbose(cmd, "Received split members from the linking object: %v", children)
+
+		// include linking object
```

Comment:
done in both cases, caller can deduplicate

Reply:
fixed